### PR TITLE
Add streaming support to listen_bucket_notification() API

### PR DESCRIPTION
**Pull request by donatello**
- Add doc on list_bucket_notification API.

Iteration style is easier to use as the code is more linear. I am not sure callback style is better suited for any particular problem (let me know if this is not right).

### CODE REVIEW

File: src/s3/client/listen_bucket_notification.rs [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+                            }
+                            let records_res: Result<NotificationRecords, Error> =
+                                serde_json::from_str(&s).map_err(|e| e.into());
+                            return Some((records_res, reader));
+                        }
+                        Err(e) => return Some((Err(e.into()), reader)),
+                    }
+                }
+            },
+        ));
```
Comment (donatello) [inline src/s3/client/listen_bucket_notification.rs:110]: Found a simpler implementation from a library for parsing lines from the http response, added as a new commit.

File: tests/tests.rs [hunk 2]
```diff
@@ -564,8 +565,15 @@ impl ClientTest {
                 false
             };
 
-            let args = &ListenBucketNotificationArgs::new(&test_bucket, &event_fn).unwrap();
-            client.listen_bucket_notification(args).await.unwrap();
+            let args = ListenBucketNotificationArgs::new(&test_bucket).unwrap();
+            let (_, mut event_stream) = client.listen_bucket_notification(args).await.unwrap();
+            while let Some(event) = event_stream.next().await {
+                // println!("event: {:?}", event);
```
Comment (balamurugana) [inline tests/tests.rs:571]: Can you remove this comment?

File: src/s3/client/listen_bucket_notification.rs
Comment (balamurugana) [inline src/s3/client/listen_bucket_notification.rs:1]: Can you add license header here?

File: tests/tests.rs (hunk 2 above)
Comment (donatello) [reply tests/tests.rs]: Done

File: src/s3/client/listen_bucket_notification.rs [hunk 3]
```diff
@@ -0,0 +1,134 @@
+// MinIO Rust Library for Amazon S3 Compatible Cloud Storage
+// Copyright 2022 MinIO, Inc.
```
Comment (balamurugana) [inline src/s3/client/listen_bucket_notification.rs:2]: Can you fix the year to `2023` for this new file?

Comment (donatello) [reply src/s3/client/listen_bucket_notification.rs]: Thank you, fixed!

**Review by balamurugana — APPROVED** [review]

**Pull request merged**